contents/javascript/Javascript-promises.md

```diff
@@ -0,0 +1,104 @@
+# Javascript: Promises
+
+Author: Daniel Berzin Chua
+
+In Javascript, there are occasions where a method call may not resolve instantly and as such, the code may not run in the sequence that you would expect due this asynchronous nature. It can be especially confusing for beginners learning Javascript.
+
+A pertinent example would be HTTP requests. When making a HTTP request, the information from the server is not instantly returned as there is a round trip time that involves the time taken for the request to be sent, and for the response from the server to be sent back to the client. In this time, Javascript would not necessarily wait for the response to be received, but instead execute other functions first, even if those functions depend on the result of that HTTP response.
+
+In the code snippet below, `console.log(price)` would return undefined as the asynchronous `retrievePrice()` function has not finished executing.
+
+```
+var price;
```

Comment:
In general for all your code examples, is it possible to color code them like in actual code? I have seen some of the resources with code examples that do this. 

This makes the code stand out more, and helps in the flow.

Reply:
Managed to do it with Markdown code syntax highlighting. Thanks for the feedback!